blockchain_connector/common/connector_common/base_connector.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import abstractmethod
+import json
+import random
+import asyncio
+import logging
+from connector_common.connector_interface \
+    import BlockchainConnectorInterface
+from avalon_sdk.connector.direct.jrpc.jrpc_worker_registry \
+    import JRPCWorkerRegistryImpl
+from avalon_sdk.connector.direct.jrpc.jrpc_work_order \
+    import JRPCWorkOrderImpl
+from connector_common.worker_delegate import \
+    WorkerDelegate
+from connector_common.work_order_delegate import \
+    WorkOrderDelegate
+
+logging.basicConfig(
+    format="%(asctime)s - %(levelname)s - %(message)s", level=logging.INFO)
+
+
+class BaseConnector(BlockchainConnectorInterface):
+
+    """
+    This abstract base class. Bridge between the blockchain and the Avalon
+    core. It listens for events generated by the blockchain.
+    It handles event data corresponding to the event (eg: workOrderSubmitted
+    and submits requests to Avalon on behalf of the client. The service also
+    invokes smart contract API (eg: workOrderComplete).
+    """
+
+    def __init__(self, config, registry_instance, worker_instance,
+                 work_order_instance, wo_receipt_instance):
+        """
+        Initialize the connector with instances of registry,
+        worker, work order and work order receipt implementation
+        classes of blockchains
+        @params
```

Comment:
Remove this line (remove @params--there is no such keyword in Doxygen). And add @param  before each param on lines 52 to 56 (for example `@param config` on line 52).

Reply:
Done